System/Metrics/Prometheus.hs

```diff
@@ -189,43 +180,39 @@ newStore = Store <$> Internal.newStore
 data MetricType
   = CounterType
   | GaugeType
-  | LabelType
-  | DistributionType
+  | HistogramType
 
 -- | The type of values sampled by each metric.
 type family MetricValue (t :: MetricType) :: Type where
-  MetricValue 'CounterType = Int64
-  MetricValue 'GaugeType = Int64
-  MetricValue 'LabelType = T.Text
-  MetricValue 'DistributionType = Distribution.Stats
+  MetricValue 'CounterType = Double
+  MetricValue 'GaugeType = Double
```

Comment:
Was there a reason to switch to double for gauges? Would this potentially lose range and precision?

Reply:
Yes, this loses range and precision, but Prometheus uses 64-bit floats for everything (https://prometheus.io/docs/introduction/faq/#why-are-all-sample-values-64-bit-floats-i-want-integers) so there's little benefit to using Ints when they get parsed as 64-bit floats in the Prometheus server. It could also give the impression of more precision/range than we actually have.